app/models/experimentation/experiment.rb

```diff
@@ -6,6 +6,7 @@ class Experiment < ActiveRecord::Base
     validates :name, presence: true, uniqueness: true
     validates :state, presence: true
     validates :experiment_type, presence: true
+    validates :experiment_type, uniqueness: true, if: proc { |experiment| experiment.experiment_type == "medication_reminder" }
```

Comment:
It seemed to me that the "find_or_create" approach you suggested was essentially enforcing uniqueness of the medication reminder type, so I chose to explicitly enforce it instead. I did this partly because there's no value in the "create" behavior because a newly created experiment would not be useful because it would not have treatment groups or reminder templates. It would prevent errors but not result in a usable experiment. This approach may not be ideal either. LMK if you have better ideas or want to brainstorm.

Reply:
This seems fine to me for now.